ballerina/websocket_service.bal

```diff
@@ -25,43 +24,64 @@ isolated service class WsService {
+            // _ = self.clientsMap.remove(caller.getConnectionId());
         }
     }
 
     isolated remote function onTextMessage(websocket:Caller caller, string data) returns websocket:Error? {
         lock {
             parser:OperationNode|ErrorDetail node = validateSubscriptionPayload(data, self.engine);
             if node is parser:OperationNode {
-                self.clientsMap[caller.getConnectionId()] = [caller, node];
-
-                stream<any, io:Error?> sourceStream = new(); // receiving stream from "getSubscriptionResponse(...)" -> Not implemented yet
-                any|io:Error? next = sourceStream.iterator().next();
-
-                while next !is io:Error? {
-                    check broadcast(self.engine, self.schema, self.context, next, self.clientsMap);
-                    next = sourceStream.iterator().next();
+                parser:Selection selection = node.getSelections()[0];
+                string methodName = "";
+                if (selection is parser:FieldNode) {
+                    methodName = selection.getName(); 
+                } else {
+                    while (selection is parser:FragmentNode) {
+                        selection = selection.getSelections()[0];
+                    } 
+                    methodName = (<parser:FieldNode> selection).getName();
                 }
+                stream<any,error?>|error sourceStream = getSubscriptionResponse(self.engine, self.schema, self.context, methodName); 
+                
+                if sourceStream is stream<any,error?>{
+                    record{|any value;|}|error? next = sourceStream.iterator().next();
+
+                    while next !is error? {
+                        ExecutorVisitor executor = new(self.engine, self.schema, self.context, {}, next.value);
+                        OutputObject outputObject = executor.getExecutorResult(node);
+                        ResponseFormatter responseFormatter = new(self.schema);
+                        OutputObject coercedOutputObject = responseFormatter.getCoercedOutputObject(outputObject, node);
+                        if coercedOutputObject.hasKey(DATA_FIELD) || coercedOutputObject.hasKey(ERRORS_FIELD) {
+                            check caller->writeTextMessage(coercedOutputObject.toString());
+                        }
+                        next = sourceStream.iterator().next();
+                    }
+                }  
+                
+            } else {
+                check caller->writeTextMessage((<ErrorDetail>node).message);
             }
-            check caller->writeTextMessage((<ErrorDetail>node).message);
+            
         }
     }
 }
 
 isolated function validateSubscriptionPayload(string text, Engine engine) returns parser:OperationNode|ErrorDetail {
     json payload = text.toJson();
     var document = payload.query;
+    if document is error {
+        document = payload;
+    }
```

Comment:
Do we need to assign payload to the `document` here?  The error is already handled.